### PR TITLE
win_acl_inheritance: Added tests and formatting improvements

### PR DESCRIPTION
**Pull request by jborean93**
##### SUMMARY
Updated win_acl_inheritance with

* check mode
* integration tests
* use types when getting arguments
* minor changes in syntax for easier viewability

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
win_acl_inheritance

##### ANSIBLE VERSION
```
ansible 2.4.0 (devel 6215922889) last updated 2017/06/06 19:40:30 (GMT +1000)
  config file = /mnt/c/dev/module-tester/ansible.cfg
  configured module search path = [u'/home/jordan/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /mnt/c/dev/linux-ansible/lib/ansible
  executable location = /mnt/c/dev/linux-ansible/bin/ansible
  python version = 2.7.13 (default, May 28 2017, 21:25:12) [GCC 5.4.0 20160609]
```

### CODE REVIEW

File: test/integration/targets/win_acl_inheritance/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,173 @@
+---
+ # Test setup
+- name: remove test folder for baseline
+  win_file:
+    path: "{{test_win_acl_inheritance_path}}"
+    state: absent
+
+- name: create test folders
+  win_file:
+    path: "{{test_win_acl_inheritance_path}}\\folder"
```
Comment (nitzmahone) [inline test/integration/targets/win_acl_inheritance/tasks/main.yml]: single-quote + no escaped backslashes (and all below)

**Review by nitzmahone — APPROVED** [review]
Looks good other than the quoting/escaping on the tests- fix it or not, up to you...

File: test/integration/targets/win_acl_inheritance/tasks/main.yml (hunk 1 above)
Comment (jborean93) [reply test/integration/targets/win_acl_inheritance/tasks/main.yml]: I thought you had to double quote a variable like "{{variable}}" if it was at the beginning unless that has changed recently?

Edit: Looks like I was wrong, you learn something new everyday. I'll update the PR accordingly as it is nicer without the escaping.

Comment (nitzmahone) [reply test/integration/targets/win_acl_inheritance/tasks/main.yml]: Yeah, you're correct that it has to be quoted if it starts the value, but single quotes work too (and solve the escaping issue nicely). So many useful little dusty corners in YAML. :D

Comment (dagwieers) [reply test/integration/targets/win_acl_inheritance/tasks/main.yml]: We need to document this. Single quotes are definitely preferred, but there is still an issue with a trailing backslash :-(

File: test/integration/test_win_group1.yml [hunk 2]
```diff
@@ -9,3 +9,4 @@
     - { role: win_fetch, tags: test_win_fetch }
     - { role: win_regmerge, tags: test_win_regmerge }
     - { role: win_regedit, tags: test_win_regedit }
+    - { role: win_acl_inheritance, tags: test_win_acl_inheritance }
```
Comment (dagwieers) [inline test/integration/test_win_group1.yml]: So this is not necessary, and even undesirable because it causes merge conflicts.

**Review by dagwieers — APPROVED** [review]

File: test/integration/test_win_group1.yml (hunk 2 above)
Comment (jborean93) [reply test/integration/test_win_group1.yml]: Yea force of habit, I'll remove it and from my other PRs

**Comment by jborean93** [comment]
Hey @nitzmahone and @dagwieers I've made the changes for this PR, let me know if you have any other changes you wish for me to make.

**Comment by jborean93** [comment]
thanks for the merge @jhawkesworth, 1 more module down :)

**Pull request merged**